net_printer/README.md

```diff
@@ -0,0 +1,116 @@
+routers consume and emit.
+
+Consider the following example output:
+
+```
+$ python3 np.py ../vtr_mcml_rwroute.phys u_calc/boundaryChecker/r_ux__57_reg[21]_srl27___u_calc_dropSpin_photon29_o_sleftz_reg_r_n_0
+============================================================
+Route tree for net: u_calc/boundaryChecker/r_ux__57_reg[21]_srl27___u_calc_dropSpin_photon29_o_sleftz_reg_r_n_0
+
+    Source: 0
+    [{      belPin  SLICE_X84Y94 C6LUT O6
+     {      sitePIP SLICE_X84Y94 OUTMUXC D6 False
+            belPin  SLICE_X84Y94 OUTMUXC OUT
+         }  sitePin SLICE_X84Y94 CMUX
+     {   }  belPin  SLICE_X84Y94 OUTMUXC D6
+            sitePin SLICE_X84Y94 C_O
+     {   }  belPin  SLICE_X84Y94 C_O C_O
+            pip     INT_X54Y94 LOGIC_OUTS_W4 INT_NODE_SDQ_49_INT_OUT1 False False
+            pip     INT_X54Y94 INT_NODE_SDQ_49_INT_OUT1 EE2_W_BEG1 False False
+            pip     INT_X55Y94 EE2_W_END1 INT_NODE_SDQ_49_INT_OUT1 False False
+            pip     INT_X55Y94 INT_NODE_SDQ_49_INT_OUT1 SS1_W_BEG1 False False
+            pip     INT_X55Y93 SS1_W_END1 INT_NODE_SDQ_50_INT_OUT1 False False
+            pip     INT_X55Y93 INT_NODE_SDQ_50_INT_OUT1 SS2_W_BEG1 False False
+            pip     INT_X55Y91 SS2_W_END1 INT_NODE_SDQ_54_INT_OUT0 False False
+            pip     INT_X55Y91 INT_NODE_SDQ_54_INT_OUT0 EE4_W_BEG1 False False
+            pip     INT_X57Y91 EE4_W_END1 INT_NODE_SDQ_53_INT_OUT1 False False
+            pip     INT_X57Y91 INT_NODE_SDQ_53_INT_OUT1 EE2_W_BEG2 False False
+            pip     INT_X58Y91 EE2_W_END2 INT_NODE_IMUX_37_INT_OUT0 False False
+            pip     INT_X58Y91 INT_NODE_IMUX_37_INT_OUT0 BYPASS_W8 True False
+            sitePin SLICE_X90Y91 EX
+            belPin  SLICE_X90Y91 EX EX
+     {   }  belPin  SLICE_X90Y91 FFMUXE1 BYP
+            sitePIP SLICE_X90Y91 FFMUXE1 BYP False
+            belPin  SLICE_X90Y91 FFMUXE1 OUT1
+         }] belPin  SLICE_X90Y91 EFF D
+============================================================
+```
+
+In this example `np.py` is invoked on the netlist `vtr_mcml_rwroute.phys` (This
+netlist that can be generated from a global `make`,
+see [the contest website](https://xilinx.github.io/fpga24_routing_contest/index.html))
+in order to printout the net named `u_calc/boundaryChecker/r_ux__57_reg[21]_srl27___u_calc_dropSpin_photon29_o_sleftz_reg_r_n_0`.
+Since this is a fully routed net the output shows one source (`Source: 0`) and
+no stubs. Following the source is a list of all of the edges (FPGAIF
+routeSegements) in the route tree. The symbol `[{` indicates root of the tree.
+The symbol `{` indicates the begining of a branch and the symbol `}` indicates
+the end of a branch. Finally, the symbol `}]` indicates the end of the tree.
+
+In the example above we see that the first edge in the tree is:
+```
+belPin  SLICE_X84Y94 C6LUT O6
+```
+which is immediately followed by a branch containing the edges:
+```
+sitePIP SLICE_X84Y94 OUTMUXC D6 False
```

Comment:
Can you explain what `False` means at the end? Maybe we should change how these are printed, or omit them entirely.

Reply:
Same for the two booleans that follow `pip` entries.

Reply:
perhaps a sentence explaining that each line simply presents all of the information contained in the [capnp routeSegment](https://github.com/chipsalliance/fpga-interchange-schema/blob/c985b4648e66414b250261c1ba4cbe45a2971b1c/interchange/PhysicalNetlist.capnp#L103)?

Reply:
In that same sentence, you could also just write out the field names.

Reply:
see: 399e8268c038d55770e1aa5077aab2c2a9c0ec9c